app/main/posts/views/filters/active-search-filters.directive.js

```diff
@@ -24,19 +35,84 @@ function ActiveSearchFilters($translate, $filter, PostFilters, _, FilterTransfor
-                return [value];
+        function handleFiltersUpdate(filters, oldValue, resetSearch) {
+            var currentFilters = angular.copy(PostFilters.getUIActiveFilters(filters));
+            FilterTransformers.rawFilters = angular.copy(filters);
+            // Remove set filter as it is only relevant to collections and should be immutable in that view
+            delete currentFilters.set;
+            var isModeSavedSearch = PostFilters.getMode() === 'savedsearch';
+            // if we are not in a saved search, make sure to reset the original and the scope saved search
+            if (!isModeSavedSearch) {
+                $scope.savedSearch = null;
+                originalSavedSearch = null;
+            }
+            /** if there is not yet a savedSearch selected or if it is a different one than the selected,
+             setup the originalSavedSearch (which NEVER changes) and the savedSearch, which changes
+             and is used for showing the filters as the user adds/removes filters
+             **/
+
+            if (resetSearch === true || !$scope.savedSearch || originalSavedSearch.id !== PostFilters.getModeId()) {
+                originalSavedSearch = angular.copy(PostFilters.getModeEntity());
+                $scope.savedSearch = PostFilters.getModeEntity();
+            }
+            /**
+             * This handles the requirement to have saved search filters displayed in a different way
+             * from the rest of the filters.
+             *  - If there is a saved search:
+             *  - - set it in the scope: this is what we will use to display the saved search and its filters in the UI
+             *  - - get clean version (no defaults) of the saved search filters
+             *  - - get a clean currentFilters array that does not include the saved search filters. $scope.uiFilters: this
+             *  is the array we will be using to show the "extra" filters the user can set AFTER they selected  a saved search
+             *  (keep in mind that when a saved search is selected all filters are erased in favor of the saved search)
+             *  - - - Diffing rules: value from currentFilters takes priority over value from search. this is because
+             *  the user will always have selected the value after they select a saved search,meaning they want to change it.
+             *  - - - Diffing rules: if the value is equal, just ignore it/send empty value.
+             *  - - - Diffing rules: return the _difference if the value is not equal, because we will want to show for instance:
+             *  saved search: tag id 1 + filters tag id 2 (so it's not just ignoring the arrays)
+             */
+            $scope.userCanUpdateSavedSearch = false;
+            $scope.uiFilters = currentFilters;
+
+            if ($scope.savedSearch) {
+                var originalSavedSearchFilters = PostFilters.getUIActiveFilters(originalSavedSearch.filter);
```

Comment:
we should be calling getUIActiveFilters way less and this whole thing should be more simple... But I need way more test coverage to be confident changing this right now TBH .  :/ It's kind of scary .